asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py

```diff
@@ -128,36 +137,326 @@ def write_html(html, path):
+        """
+        Generated using PLIP docs; colors match PyMol interaction colors. See
+        https://github.com/pharmai/plip/blob/master/DOCUMENTATION.md
+        converted RGB to HEX using https://www.rgbtohex.net/
+        """
+        if intn_type == "hydrophobic_interaction":
+            return "#808080"
+        elif intn_type == "hydrogen_bond":
+            return "#0000FF"
+        elif intn_type == "water_bridge":
+            return "#BFBFFF"
+        elif intn_type == "salt_bridge":
+            return "#FFFF00"
+        elif intn_type == "pi_stack":
+            return "#00FF00"
+        elif intn_type == "pi_cation_interaction":
+            return "#FF8000"
+        elif intn_type == "halogen_bond":
+            return "#36FFBF"
+        elif intn_type == "metal_complex":
+            return "#8C4099"
+        else:
+            raise ValueError(f"Interaction type {intn_type} not recognized.")
+
+    def build_interaction_dict(self, plip_xml_dict, intn_counter, intn_type) -> Union:
+        """
+        Parses a PLIP interaction dict
+        """
+        k = f"{intn_counter}_{plip_xml_dict['restype']}{plip_xml_dict['resnr']}.{plip_xml_dict['reschain']}"
+
+        v = {
+            "lig_at_x": plip_xml_dict["ligcoo"]["x"],
+            "lig_at_y": plip_xml_dict["ligcoo"]["y"],
+            "lig_at_z": plip_xml_dict["ligcoo"]["z"],
+            "prot_at_x": plip_xml_dict["protcoo"]["x"],
+            "prot_at_y": plip_xml_dict["protcoo"]["y"],
+            "prot_at_z": plip_xml_dict["protcoo"]["z"],
+            "type": intn_type,
+            "color": self.get_interaction_color(intn_type),
+        }
+        return k, v
+
+    @staticmethod
+    def get_interactions_plip(self, pose) -> dict:
+        """
+        Get protein-ligand interactions according to PLIP.
+
+        TODO:
+        currently this uses a tmp PDB file, uses PLIP CLI (python package throws
+        ```
+        libc++abi: terminating with uncaught exception of type swig::stop_iteration
+        Abort trap: 6
+        ```
+        ), then parses XML to get interactions. This is a bit convoluted, we could refactor
+        this to use OE's InteractionHints instead? ProLIF struggles to detect incorrections
+        because it's v sensitive to protonation. PLIP does protonation itself.
+        """
+        # create the complex PDB file.
+        save_openeye_pdb(combine_protein_ligand(self.protein, pose), "tmp_complex.pdb")
+
+        # run the PLIP CLI.
+        subprocess.run(["plip", "-f", "tmp_complex.pdb", "-x", "-o", "tmp"])
+
+        # load the XML produced by PLIP that contains all the interaction data.
+        intn_dict_xml = xmltodict.parse(
+            ET.tostring(ET.parse("tmp/report.xml").getroot())
+        )
+
+        intn_dict = {}
+        intn_counter = 0
+        # wrangle all interactions into a dict that can be read directly by 3DMol.
+        for _, data in intn_dict_xml["report"]["bindingsite"]["interactions"].items():
```

Comment:
Tests currently choke on this: looks like `intn_dict_xml["report"]["bindingsite"]` is a `list`, with more than one item. Not sure what the correct thing to do here is; do we expect more than one binding site in many cases?

Reply:
ah, that might be happening if we have two ligands bound to a dimer (in case of e..g Mpro). In principle that's fine, we can show interactions on both monomers. I'll change this to account for that.